cells/src/commonTest/kotlin/com/wire/kalium/cells/domain/usecase/AddAttachmentDraftUseCaseTest.kt

```diff
@@ -0,0 +1,239 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.cells.domain.usecase
+
+import com.wire.kalium.cells.domain.CellConversationRepository
+import com.wire.kalium.cells.domain.CellUploadEvent
+import com.wire.kalium.cells.domain.CellUploadManager
+import com.wire.kalium.cells.domain.MessageAttachmentDraftRepository
+import com.wire.kalium.cells.domain.model.AttachmentUploadStatus
+import com.wire.kalium.cells.domain.model.CellNode
+import com.wire.kalium.common.functional.right
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.AssetContent
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.coEvery
+import io.mockative.coVerify
+import io.mockative.every
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runTest
+import okio.Path.Companion.toPath
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class AddAttachmentDraftUseCaseTest {
+
+    private companion object {
+        val conversationId = ConversationId("1", "test")
+        val assetPath = "path".toPath()
+        const val assetSize = 1000L
+        const val fileName = "testfile.test"
+        val destNodePath = "wire-cells-android/$conversationId/$fileName"
+        val mimeType = "image/jpeg"
+        val metadata = AssetContent.AssetMetadata.Image(0, 0)
+    }
+
+    @Test
+    fun given_valid_request_upload_manager_is_called() = runTest {
+        val (arrangement, useCase) = Arrangement()
+            .withSuccessAdd()
+            .withSuccessPreCheck()
+            .arrange()
+
+        useCase(conversationId, fileName, mimeType, assetPath, assetSize, metadata)
+
+        coVerify {
+            arrangement.uploadManager.upload(
+                assetPath = assetPath,
+                assetSize = assetSize,
+                destNodePath = destNodePath,
+            )
+        }.wasInvoked(once)
+    }
+
+    @Test
+    fun given_success_pre_check_attachment_is_persisted() = runTest {
```

Comment:
Nit, non blocker.

I personally tend to find the standard `givenSomeState_whenSomethingHappens_thenResultShouldBe` a good bit more readable than `given_some_state_when_something_happens_then_result_should_be`, when we're glancing over the scenarios and trying to find specific `thens` or `givens` within the test class.